rules/precompiled_apple_resource_bundle.bzl

```diff
@@ -0,0 +1,235 @@
+only once
+NOTE: This rule only exists because of this issue
+https://github.com/bazelbuild/rules_apple/issues/319
+if this is ever fixed in bazel it should be removed
+"""
+
+load("@bazel_skylib//lib:partial.bzl", "partial")
+load("@bazel_skylib//lib:paths.bzl", "paths")
+load("@build_bazel_rules_apple//apple/internal:apple_product_type.bzl", "apple_product_type")
+load("@build_bazel_rules_apple//apple/internal:intermediates.bzl", "intermediates")
+load("@build_bazel_rules_apple//apple/internal:partials.bzl", "partials")
+load("@build_bazel_rules_apple//apple/internal:resource_actions.bzl", "resource_actions")
+load("@build_bazel_rules_apple//apple/internal:rule_factory.bzl", "rule_factory")
+load("//rules:transition_support.bzl", "transition_support")
+load("@build_bazel_rules_apple//apple:providers.bzl", "AppleResourceBundleInfo", "AppleResourceInfo")
+
+_FAKE_BUNDLE_PRODUCT_TYPE_BY_PLATFORM_TYPE = {
+    "ios": apple_product_type.application,
+    "tvos": apple_product_type.application,
+    "watchos": apple_product_type.watch2_application,
+}
+
+def _precompiled_apple_resource_bundle_impl(ctx):
+    bundle_name = ctx.attr.bundle_name or ctx.label.name
+
+    attr_dict = {}
+    for k in dir(ctx.attr):
+        if k in ("to_json", "to_proto"):
+            continue
+        attr_dict[k] = getattr(ctx.attr, k)
+
+    current_apple_platform = transition_support.current_apple_platform(apple_fragment = ctx.fragments.apple, xcode_config = ctx.attr._xcode_config)
+    attr_dict["platform_type"] = str(current_apple_platform.platform.platform_type)
+    attr_dict["minimum_os_version"] = str(current_apple_platform.target_os_version)
+    attr_dict["_product_type"] = _FAKE_BUNDLE_PRODUCT_TYPE_BY_PLATFORM_TYPE.get(attr_dict["platform_type"], ctx.attr._product_type)
+
+    file_dict = {}
+    for k in dir(ctx.file):
+        if k in ("to_json", "to_proto"):
+            continue
+        file_dict[k] = getattr(ctx.file, k)
+
+    file_dict["_environment_plist"] = [f for f in ctx.files._environment_plists if f.path.endswith("_{}.plist".format(attr_dict["platform_type"]))][0]
+
+    fake_ctx_dict = {}
+    for k in dir(ctx):
+        if k in ("aspect_ids", "build_setting_value", "rule", "to_json", "to_proto"):
+            continue
+        fake_ctx_dict[k] = getattr(ctx, k)
+    fake_ctx_dict["attr"] = struct(**attr_dict)
+    fake_ctx_dict["file"] = struct(**file_dict)
+    fake_ctx = struct(**fake_ctx_dict)
+
+    partial_output = partial.call(partials.resources_partial(
+        top_level_attrs = ["resources"],
+    ), fake_ctx)
+
+    # Process the plist ourselves. This is required because
+    # include_executable_name defaults to True, which results in iTC rejecting
+    # our binary.
+    output_plist = ctx.actions.declare_file(
+        paths.join("%s-intermediates" % ctx.label.name, "Info.plist"),
+    )
+    bundle_id = ctx.attr.bundle_id or "com.cocoapods." + bundle_name
```

Comment:
Would it make sense to make the `bundle_id` required instead of providing a default?

Reply:
Maybe? We could push the default out to the `apple_library` macro if you think that's better, but we do need a default at some layer